src/components/TokenBalanceChecker.js

```diff
@@ -0,0 +1,342 @@
+		super(props);
+
+		this.state = {
+			error_message_class: "",
+			token_balance_error_message: "",
+			token_balance_container_class: "",
+			check_balance_button_spinner_class: "hide",
+			check_balance_button_text_class: "",
+			selectedTokenByUser: false,
+			selectedToken: "",
+			token_balance_response: false,
+			check_balance_button_text: "",
+			capture_token_balance_container_input_id: "",
+			capture_token_balance_container_input_type: "",
+			capture_token_balance_input_placeholder: "",
+			capture_token_balance_input_value: ""
+		};
+	}
+
+	tokenChange = (e) => {
+		this.setState({
+			token_balance_response: false,
+			error_message_class: "",
+			token_balance_container_class: "show",
+			capture_token_balance_input_value: "",
+			selectedTokenByUser: true
+		});
+
+		let tokenSymbol = e.target.value;
+		let button_text = "";
+		let balance_container_input_id = "";
+		let balance_container_input_type = "";
+		let input_placeholder = "";
+		let token_balance_input_value = "";
+
+		if (tokenSymbol === "0") {
+			this.setState({
+				selectedTokenByUser: false,
+				token_balance_container_class: ""
+			});
+			return
+		} else {
+			if (tokenSymbol === "all") {
+				button_text = "Check";
+				balance_container_input_id = "token_balance_wallet_address";
+				balance_container_input_type = "text";
+				input_placeholder = "Enter BEP-20 Public Wallet Address";
+				let public_wallet_address = Cookies.get('public_wallet_address');
+				if (public_wallet_address != undefined) {
+					token_balance_input_value = public_wallet_address;
+				}
+			} else {
+				button_text = "Calculate";
+				balance_container_input_id = "token_balance";
+				balance_container_input_type = "number";
+				input_placeholder = "Enter "+tokenSymbol+" Amount";
+				let selected_token_amount = Cookies.get(tokenSymbol+'_token_amount');
+				if (selected_token_amount != undefined) {
+					token_balance_input_value = selected_token_amount;
+				}
+			}
+
+			this.setState({
+				check_balance_button_text: button_text,
+				capture_token_balance_container_input_id: balance_container_input_id,
+				capture_token_balance_container_input_type: balance_container_input_type,
+				selectedToken: tokenSymbol,
+				capture_token_balance_input_placeholder: input_placeholder,
+				capture_token_balance_input_value: token_balance_input_value
+			});
+		}
+	}
+
+	renderCheckTokenBalancesView = () => {
+		if (this.state.selectedTokenByUser && this.state.selectedToken !== "0") {
+			return true;
+		}
+		
+		return false;
+	}
+
+	lookupBalances = (ev) => {
+		// Reset states
+		this.setState({
+			error_message_class: "",
+			check_balance_button_spinner_class: "",
+			check_balance_button_text_class: "hide"
+		});
+		
+		let promises = [];
+		let wallet_response = {};
+		let tokens_to_check = [];
+		let formated_wallet_address = "";
+
+		if (this.state.selectedToken == "all") {
+			let wallet_address = document.getElementById('token_balance_wallet_address');
```

Comment:
slightly bad practice for React stuff. You should probably be updating a state when switching, then pulling this information from that state rather than from an HTMLElement Id.

Reply:
done